packages/e2e/test/web-extension/extension/background/config.ts

```diff
@@ -9,50 +11,65 @@ import {
+  rewardsHttpProvider,
+  txSubmitHttpProvider,
+  utxoHttpProvider
+} from '@cardano-sdk/cardano-services-client';
+import { createConnectionObject } from '@cardano-ogmios/client';
 import { createStubStakePoolProvider } from '@cardano-sdk/util-dev';
+import { ogmiosTxSubmitProvider } from '@cardano-sdk/ogmios';
 import axiosFetchAdapter from '@vespaiach/axios-fetch-adapter';
 
-const loggerMethodNames = ['debug', 'error', 'fatal', 'info', 'trace', 'warn'] as (keyof Logger)[];
 const networkIdOptions = [0, 1];
-const stakePoolProviderOptions = ['stub'];
-const networkInfoProviderOptions = ['blockfrost'];
-const txSubmitProviderOptions = ['blockfrost'];
-const assetProviderOptions = ['blockfrost'];
-const utxoProviderOptions = ['blockfrost'];
-const rewardsProviderOptions = ['blockfrost'];
-const keyAgentOptions = ['InMemory'];
-const chainHistoryProviderOptions = ['blockfrost'];
+const stakePoolProviderOptions = ['stub', 'http'];
+const networkInfoProviderOptions = ['blockfrost', 'http'];
+const txSubmitProviderOptions = ['blockfrost', 'http'];
+const assetProviderOptions = ['blockfrost', 'http'];
+const utxoProviderOptions = ['blockfrost', 'http'];
+const rewardsProviderOptions = ['blockfrost', 'http'];
+const chainHistoryProviderOptions = ['blockfrost', 'http'];
 
 const env = envalid.cleanEnv(process.env, {
   ASSET_PROVIDER: envalid.str({ choices: assetProviderOptions }),
+  ASSET_PROVIDER_PARAMS: envalid.json({ default: {} }),
   BLOCKFROST_API_KEY: envalid.str(),
   CHAIN_HISTORY_PROVIDER: envalid.str({ choices: chainHistoryProviderOptions }),
-  KEY_AGENT: envalid.str({ choices: keyAgentOptions }),
-  LOGGER_MIN_SEVERITY: envalid.str({ choices: loggerMethodNames as string[], default: 'info' }),
+  CHAIN_HISTORY_PROVIDER_PARAMS: envalid.json({ default: {} }),
+  LOGGER_MIN_SEVERITY: envalid.str({ default: 'info' }),
   MNEMONIC_WORDS: envalid.makeValidator<string[]>((input) => {
     const words = input.split(' ');
     if (words.length === 0) throw new Error('MNEMONIC_WORDS not set');
     return words;
   })(),
   NETWORK_ID: envalid.num({ choices: networkIdOptions }),
   NETWORK_INFO_PROVIDER: envalid.str({ choices: networkInfoProviderOptions }),
-  POOL_ID_1: envalid.str(),
-  POOL_ID_2: envalid.str(),
+  NETWORK_INFO_PROVIDER_PARAMS: envalid.json({ default: {} }),
   REWARDS_PROVIDER: envalid.str({ choices: rewardsProviderOptions }),
+  REWARDS_PROVIDER_PARAMS: envalid.json({ default: {} }),
   STAKE_POOL_PROVIDER: envalid.str({ choices: stakePoolProviderOptions }),
-  TX_SUBMIT_HTTP_URL: envalid.url(),
+  STAKE_POOL_PROVIDER_PARAMS: envalid.json({ default: {} }),
   TX_SUBMIT_PROVIDER: envalid.str({ choices: txSubmitProviderOptions }),
+  TX_SUBMIT_PROVIDER_PARAMS: envalid.json({ default: {} }),
   UTXO_PROVIDER: envalid.str({ choices: utxoProviderOptions }),
-  WALLET_PASSWORD: envalid.str()
+  UTXO_PROVIDER_PARAMS: envalid.json({ default: {} })
 });
+
+const logger = {
+  debug: (x: any) => console.log(x),
```

Comment:
Might be called with multiple args
```suggestion
  debug: (...x: any[]) => console.log(...x),
```
What is the reason for this change though (why not just use `logger = console`)?

Reply:
The extension does not log debug messages if I just do` logger = console` maybe it is the way the extension is being compiled?

Reply:
It does log debug messages, you just need to unfilter it in 'Default levels'

Reply:
the changes to the core packages were split into a different commit here:
https://github.com/input-output-hk/cardano-js-sdk/pull/356/commits/a38fe7dfbee2889497aeb42f5b1847a0c02720e4